Support onprem platform for user-tools profiling command

nartal1 opened the pull request:
This PR fixes https://github.com/NVIDIA/spark-rapids-tools/issues/315.
This PR adds user-tools profiling command for On-premisis platform. Updated the documentation for profiling CLI.

Command to test profiling tool:

```
export EVENT_LOGS=$SCRATCH_DIR/eventlogs
export WORKER_INFO=$SCRATCH_DIR/gpu_worker_info.yaml

 spark_rapids_user_tools onprem profiling \
    --eventlogs $EVENT_LOGS \
    --worker_info $WORKER_INFO
```


<!--

Thank you for contributing to Tools of RAPIDS Accelerator for Apache Spark!

Here are some guidelines to help the review process go smoothly.

1. Please write a description in this text box of the changes that are being
   made.

2. Please ensure that you have written units tests for the changes made/features
   added.

3. If you are closing an issue please use one of the automatic closing words as
   noted here: https://help.github.com/articles/closing-issues-using-keywords/

4. If your pull request is not ready for review but you want to make use of the
   continuous integration testing facilities please label it with `[WIP]`.

5. If your pull request is ready to be reviewed without requiring additional
   work on top of it, then remove the `[WIP]` label (if present).

6. Once all work has been done and review has taken place please do not add
   features or make changes out of the scope of those requested by the reviewer
   (doing this just add delays as already reviewed code ends up having to be
   re-reviewed/it is hard to tell what is new etc!). Further, please avoid
   rebasing your branch during the review process, as this causes the context
   of any comments made by reviewers to be lost. If conflicts occur during
   review then they should be resolved by merging into the branch used for
   making the pull request.

Many thanks in advance for your cooperation!

-->

mattahrens commented on user_tools/docs/user-tools-onprem.md:
```diff
@@ -219,4 +219,95 @@ The command creates a directory with UUID that contains the following:
+| **verbose**          | True or False to enable verbosity to the wrapper script                                                                                                                                                                                                   | False if `RAPIDS_USER_TOOLS_LOG_DEBUG` is not set                                                                          ...
+| **rapids_options**** | A list of valid [Profiling tool options](../../core/docs/spark-profiling-tool.md#qualification-tool-options). Note that (`output-directory`, `auto-tuner`, `combined`) flags are ignored                                                                  | N/A                                                                                                                        ...
+
+If the CLI does not provide an argument `worker_info`, the tool will throw an error and exit.
```
This is fine for now.  But would it be simple to allow the user to run profiling without worker_info and then not execute auto-tuner and just generate profiling output files locally?  That would be needed to fully deprecate Profiler java command as some customers have run w/o cluster info to get metrics output without recommendations.

amahussein replied:
If that's the case, then I suggest that to be done within the same PR so we don't have to go back to the `spark_rapids_user_tools` CLI to add features.
The trick with the new `ascli` proposal that it recommends removing the "rapids_options" arguments which means that some arguments consumed by the jar file have to be hardcoded.

amahussein reviewed (commented):
- @nartal1 If you find that it is easy modifying the profiling command to only run the java command without recommendation, then we can do it within the same PR. Otherwise, then we can keep that feature out of scope since it is part of the proposal of the new CLI
- Something that was overlooked in previous PRs is that onPRem qualification does not have doc-string to generate the help command. We can add the doc-string of the qualification here to get done with it.

nartal1 commented:
> * @nartal1 If you find that it is easy modifying the profiling command to only run the java command without recommendation, then we can do it within the same PR. Otherwise, then we can keep that feature out of scope since it is part of the proposal of the new CLI
> * Something that was overlooked in previous PRs is that onPRem qualification does not have doc-string to generate the help command. We can add the doc-string of the qualification here to get done with it.

Modifying profiling command to run java command without recommendation will take more effort. As discussed offline, we will incorporate this change in the new CLI.

Updated the doc-string for qualification command. Removed arguments which are not needed for on prem platform in that process. PTAL
